junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/TestExecutionSummary.java

```diff
@@ -127,4 +129,16 @@
 	 */
 	void printFailuresTo(PrintWriter writer);
 
+	/**
+	 * Get a list of the failures of this test execution.
+	 */
+	List<Failure> getFailures();
+
+	interface Failure {
```

Comment:
Since this is now a public type, both the interface and its methods require proper Javadoc.